azerothcore/azerothcore-wotlk
# refactor(Scripts/ShatteredHalls): move Porung from SAI to cpp

## Changes Proposed:
-  Move boss script from SAI to C++.
-  Clean-up db data for unusable action timed list for Porung and his normal version counterpart.
-  Reset the scheduled actions from Shattered Hand Scout once the boss is killed (normal too)
-  Move the gauntlet event from instance_shattered_halls to porung script.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/16072
(it should be closed already but well, this is related too)

## SOURCE:


## Tests Performed:

- Tested in-game, normal and heroic.

## How to Test the Changes:


1. Go to shattered halls, kill Nethekurse and go through the door.
2. Once you spot the Scout, he should run and start the gauntlet.
3. Everything should work as before.
